From 65b48186c49363658731da38a693b3f7c80a7619 Mon Sep 17 00:00:00 2001
From: Jianing Ren <jianing.ren@rock-chips.com>
Date: Mon, 1 Jun 2020 20:00:54 +0800
Subject: [PATCH] phy: phy-rockchip-naneng-usb2: cancel the dynamic management
 of clks

This patch remove the clks manegement in phy operations and keep the
clks always on. Keep the clks on can avoid many errors such as USB480M
clk abnormal output and EHCI controller error.

Change-Id: If201106d432d05e8fc5f63d595a67524e2183b5c
Signed-off-by: Jianing Ren <jianing.ren@rock-chips.com>
---
 .../phy/rockchip/phy-rockchip-naneng-usb2.c   | 28 +++----------------
 1 file changed, 4 insertions(+), 24 deletions(-)

diff --git a/drivers/phy/rockchip/phy-rockchip-naneng-usb2.c b/drivers/phy/rockchip/phy-rockchip-naneng-usb2.c
index eecda6bba62f..6bc5c64482c0 100644
--- a/drivers/phy/rockchip/phy-rockchip-naneng-usb2.c
+++ b/drivers/phy/rockchip/phy-rockchip-naneng-usb2.c
@@ -569,23 +569,15 @@ static int rockchip_usb2phy_power_on(struct phy *phy)
 		goto unlock;
 	}
 
-	ret = clk_bulk_enable(rphy->num_clks, rphy->clks);
-	if (ret)
-		goto unlock;
-
 	ret = property_enable(rphy->grf, &rport->port_cfg->phy_sus, false);
 	if (ret)
-		goto disable_clks;
+		goto unlock;
 
 	/* waiting for the utmi_clk to become stable */
 	usleep_range(2500, 3000);
 
 	rport->suspended = false;
 
-	goto unlock;
-
-disable_clks:
-	clk_bulk_disable(rphy->num_clks, rphy->clks);
 unlock:
 	mutex_unlock(&rport->mutex);
 
@@ -611,8 +603,6 @@ static int rockchip_usb2phy_power_off(struct phy *phy)
 	if (ret)
 		goto unlock;
 
-	clk_bulk_disable(rphy->num_clks, rphy->clks);
-
 	rport->suspended = true;
 
 unlock:
@@ -759,6 +749,9 @@ static void rockchip_chg_detect(struct rockchip_usb2phy *rphy,
 
 	usleep_range(1000, 1100);
 	reset_control_deassert(rphy->reset);
+	/* waiting for the utmi_clk to become stable */
+	usleep_range(2500, 3000);
+
 	/* disable the chg detection module */
 	property_enable(rphy->grf, &rphy->phy_cfg->chg_det.chg_rst, true);
 	property_enable(rphy->grf, &rphy->phy_cfg->chg_det.chg_en, false);
@@ -846,16 +839,8 @@ static void rockchip_usb2phy_otg_sm_work(struct work_struct *work)
 		if (rport->perip_connected)
 			return;
 
-		if (clk_bulk_enable(rphy->num_clks, rphy->clks))
-			return;
-
-		/* waiting for the utmi_clk to become stable */
-		usleep_range(2500, 3000);
-
 		rockchip_chg_detect(rphy, rport);
 
-		clk_bulk_disable(rphy->num_clks, rphy->clks);
-
 		switch (rphy->chg_type) {
 		case POWER_SUPPLY_TYPE_USB:
 			dev_dbg(&rport->phy->dev, "sdp cable is connected\n");
@@ -1351,11 +1336,6 @@ static int rockchip_usb2phy_probe(struct platform_device *pdev)
 	else
 		device_init_wakeup(rphy->dev, false);
 
-	/*
-	 * Let us disable clks here for saving power consumption, and usb
-	 * controller will resume it during probe time if needed.
-	 */
-	clk_bulk_disable(rphy->num_clks, rphy->clks);
 	return 0;
 
 put_child:
-- 
2.35.3

